### ISSUE TITLE
[Feature Request] option to remap lighten/darken tool to two-button mouse

### ISSUE THREAD

**Issue by johnthehappyjohn** [piskelapp/piskel#972]
Suggest allowing L & R clicks to be mapped to separate functions (Lighten/darken) respectively. Could also apply to select tool (L to move select field, R to move selected contents) reducing required key combinations.

**Comment by OPStellar** [comment]
Hi, johnthehappyjohn,
There's a little problem with your suggestion. You specify that secondary clicks would move selected contents, but this can already be accomplished by using the 'Shift' key. The command/Ctrl key can be use similarly to darken with the lighten/darken tool.
Hope this helps!
Opie

**Comment by johnthehappyjohn** [comment]
Many thanks!

I am aware; however, both right and left clicks do the same thing with those two functions. Seems odd to require a keyboard combination when there's a perfectly viable mouse button crying out for a secondary function (altfire!) as it is for the brush tool. I am unable to bind specific functions of tools to the mouse buttons; only to select the tool itself, and then, only to keyboard inputs.

Hope this clarifies.

JoHaJo
________________________________

**Comment by OPStellar** [comment]
You taught me a new power. Piskel surprises me every time I use it.
I had forgotten it could even do secondary clicks! I probably stopped doing that when I learned that I could just use the x shortcut key and focus on positioning of my clicks without worrying whether my use of two fingers might redirect the placement (I use Piskel on a MacBook Pro with a trackpad, and I don't use the 'control+click' method for secondary clicks).

That said, I think that the way it is mapped is strictly to color. I get what you mean about how that would be helpful in lightening and darkening; however, that's brightness, not hue, and the shift key does what you're requesting in that regard. Understand that as soon as there is more than one function for a primary action (such as clicking and secondary clicking), bugs will arise.
Also, as a trackpad user, I don't find it very easy to draw with two fingers on the trackpad at once… so just for the sake of universal design, I don't think mapping more functions to secondary clicks would be a step in the right direction.
There are alternatives, though, in the area of making selections:
There is a request for a 'select all pixels of the same color' expansion to the shape selection tool. Perhaps a key command could be used while making selections that will invert the selection? I'm not sure how practical that would be, but it might be worth exploring.

Anyway, here's the tl:dr…
clicking < typing
Most fluent users prefer to use the modifier keys (command/ctrl, alt/option, shift, etc.) to perform tasks than to click.
I recommend you request a non-click option. Thanks for your consideration of my feedback.

**Comment by johnthehappyjohn** [comment]
Fair enough; this might become less necessary over time with familiarity. As you say, the modifier keys are already there and functional! An 'invert selection' modifier is an excellent shout though!! Perhaps it's just time to shell out for Aseprite...As I've no points in coding, I can't appreciate the inevitable bugs I'm probably implying. Thanks anyway!JoHaJoOn 25 Oct 2020 16:05, Sir Vyvre the Lone <notifications@github.com> wrote:

You taught me a new power. Piskel surprises me every time I use it.

I had forgotten it could even do secondary clicks! I probably stopped doing that when I learned that I could just use the x shortcut key and focus on positioning of my clicks without worrying whether my use of two fingers might redirect the placement (I use Piskel on a MacBook Pro with a trackpad, and I don't use the 'control+click' method for secondary clicks).



That said, I think that the way it is mapped is strictly to color. I get what you mean about how that would be helpful in lightening and darkening; however, that's brightness, not hue, and the shift key does what you're requesting in that regard. Understand that as soon as there is more than one function for a primary action (such as clicking and secondary clicking), bugs will arise.

Also, as a trackpad user, I don't find it very easy to draw with two fingers on the trackpad at once… so just for the sake of universal design, I don't think mapping more functions to secondary clicks would be a step in the right direction.

There are alternatives, though, in the area of making selections:

There is a request for a 'select all pixels of the same color' expansion to the shape selection tool. Perhaps a key command could be used while making selections that will invert the selection? I'm not sure how practical that would be, but it might be worth exploring.



Anyway, here's the tl:dr…

clicking < typing

Most fluent users prefer to use the modifier keys (command/ctrl, alt/option, shift, etc.) to perform tasks than to click.

I recommend you request a non-click option. Thanks for your consideration of my feedback.

**Comment by OPStellar** [comment]
I'll stick with Piskel, it's a solid software that works on Mac. I don't know what updates have come out lately, but I trust that some are in the works! Hope you come back soon!
